homeassistant/components/advantage_air/sensor.py

```diff
@@ -0,0 +1,212 @@
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "On"))
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "Off"))
+        for zone_key, zone in ac_device["zones"].items():
+            # Only show damper sensors when zone is in temperature control
+            if zone["type"] != 0:
+                entities.append(AdvantageAirZoneVent(instance, ac_key, zone_key))
+            # Only show wireless signal strength sensors when using wireless sensors
+            if zone["rssi"] > 0:
+                entities.append(AdvantageAirZoneSignal(instance, ac_key, zone_key))
+    async_add_entities(entities)
+
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        ADVANTAGE_AIR_SERVICE_SET_TIME_TO,
+        {vol.Required("minutes"): cv.positive_int},
+        "set_time_to",
+    )
+
+
+class AdvantageAirTimeTo(CoordinatorEntity):
+    """Representation of Advantage Air timer control."""
+
+    def __init__(self, instance, ac_key, time_period):
+        """Initialize the Advantage Air timer control."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.ac_key = ac_key
+        self.time_period = time_period
+
+    @property
+    def _ac(self):
+        return self.coordinator.data["aircons"][self.ac_key]["info"]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f'{self._ac["name"]} Time To {self.time_period}'
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-timeto{self.time_period}'
+
+    @property
+    def state(self):
+        """Return the current value."""
+        return self._ac[f"countDownTo{self.time_period}"]
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return ADVANTAGE_AIR_SET_COUNTDOWN_UNIT
+
+    @property
+    def icon(self):
+        """Return a representative icon of the timer."""
+        if self._ac[f"countDownTo{self.time_period}"] > 0:
+            return "mdi:timer-outline"
+        return "mdi:timer-off-outline"
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return {
+            "identifiers": {(DOMAIN, self.coordinator.data["system"]["rid"])},
+            "name": self.coordinator.data["system"]["name"],
+            "manufacturer": "Advantage Air",
+            "model": self.coordinator.data["system"]["sysType"],
+            "sw_version": self.coordinator.data["system"]["myAppRev"],
+        }
+
+    async def set_time_to(self, **kwargs):
+        """Set the timer value."""
+        value = min(720, max(0, int(kwargs[ADVANTAGE_AIR_SET_COUNTDOWN_VALUE])))
+        await self.async_change(
+            {self.ac_key: {"info": {f"countDownTo{self.time_period}": value}}}
+        )
+
+
+class AdvantageAirZoneVent(CoordinatorEntity):
```

Comment:
Looks like you start to get a common ground between your entities you could break out to a base class